homeassistant/components/lutron/config_flow.py

```diff
@@ -0,0 +1,89 @@
+"""Config flow to configure the Lutron integration."""
+from __future__ import annotations
+
+import logging
+from urllib.error import HTTPError
+
+from pylutron import Lutron
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LutronConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """User prompt for Main Repeater configuration information."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """First step in the config flow."""
+        errors = {}
+
+        if user_input is not None:
+            # Validate user input
+            ip_address = user_input[CONF_HOST]
+
+            # Now that we have the data, check and see if we can connect and get a GUID
+            main_repeater = Lutron(
+                ip_address,
+                user_input.get(CONF_USERNAME),
+                user_input.get(CONF_PASSWORD),
+            )
+
+            def _load_db() -> bool:
+                main_repeater.load_xml_db()
+                return True
+
+            try:
+                await self.hass.async_add_executor_job(_load_db)
+                guid = main_repeater.guid
+            except UnknownConnectError:
+                errors["base"] = "config_errors"
+            except HTTPError as ex:
+                # In a future version we can get more specific with the HTTP code
+                _LOGGER.debug("Exception Type: %s", type(ex).__name__)
+                errors["base"] = "connect_error"
+
+            if not errors:
+                if len(guid) > 10:
+                    _LOGGER.info("Main Repeater GUID: %s", main_repeater.guid)
+                else:
+                    errors["base"] = "missing_guid"
+
+            # Check if a configuration entry with the same unique ID already exists
+            if not errors:
+                if self._async_current_entries():
+                    return self.async_abort(reason="single_instance_allowed")
```

Comment:
Just reminded myself we can place this at the top of this function, as we can abort the flow directly when setting up. (otherwise the user can still enter details, and then get aborted because it already existed (but we could already know this))

Reply:
moved to top